src/lex/mod.rs

```diff
@@ -374,31 +373,21 @@ impl Lexer {
             self.next_char();
         }
         let float = if hex {
```

Comment:
You can avoid this assignment + `?` conversion by having the `if` be the last expression in the block:

```rust
if hex {
    parse_hexf64().map_err()
} else {
    buf.parse()
}

Reply:
Can't because of the underflow check.

Reply:
Ah right, mutually exclusive review comments :laughing: 

offtopic: `rustfix` actually had this exact same issue: https://discordapp.com/channels/442252698964721669/446773253953028107/651615006898782233